pootle/apps/pootle_store/diff.py

```diff
@@ -51,32 +51,37 @@ def target(self):
 
 class StoreDiff(object):
 
-    def __init__(self, db_store, file_store, file_revision):
-        self.db_store = db_store
-        self.file_store = file_store
-        self.db_revision = db_store.get_max_unit_revision()
-        self.file_revision = file_revision
+    def __init__(self, target_store, source_store, source_revision):
+        self.target_store = target_store
+        self.source_store = source_store
+        self.source_revision = source_revision
+
+    @cached_property
+    def target_revision(self):
+        return self.target_store.get_max_unit_revision()
```

Comment:
IIRC we should get db_revision before we start update process to avoid any of race conditions.
I'm not clear when we call this method first time in the current implementation.